File: Models/Stics_soil_temperature/src/java/Stics_soil_temperature/soil_tempAuxiliary.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class soil_tempAuxiliary
{
    private Double temp_wave_freq;
    private Double therm_diff;
    
    public soil_tempAuxiliary() { }
    
    public soil_tempAuxiliary(soil_tempAuxiliary toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.temp_wave_freq = toCopy.gettemp_wave_freq();
            this.therm_diff = toCopy.gettherm_diff();
        }
    }
    public Double gettemp_wave_freq()
    { return temp_wave_freq; }

    public void settemp_wave_freq(Double _temp_wave_freq)
    { this.temp_wave_freq= _temp_wave_freq; } 
    
    public Double gettherm_diff()
    { return therm_diff; }

    public void settherm_diff(Double _therm_diff)
    { this.therm_diff= _therm_diff; } 
    
}

File: Models/Stics_soil_temperature/src/cpp/Stics_soil_temperature/soil_tempAuxiliary.cpp
#include "soil_tempAuxiliary.h"

soil_tempAuxiliary::soil_tempAuxiliary() { }

double soil_tempAuxiliary::gettemp_wave_freq() {return this-> temp_wave_freq; }
double soil_tempAuxiliary::gettherm_diff() {return this-> therm_diff; }

void soil_tempAuxiliary::settemp_wave_freq(double _temp_wave_freq) { this->temp_wave_freq = _temp_wave_freq; }
void soil_tempAuxiliary::settherm_diff(double _therm_diff) { this->therm_diff = _therm_diff; }